dev/NumberBox/NumberBox.cpp

```diff
@@ -309,88 +384,143 @@ void NumberBox::OnNumberBoxKeyUp(winrt::IInspectable const& sender, winrt::KeyRo
     }
 }
 
-void NumberBox::OnScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args)
+void NumberBox::OnNumberBoxScroll(winrt::IInspectable const& sender, winrt::PointerRoutedEventArgs const& args)
 {
-    if (HyperScrollEnabled())
+    if (auto && textBox = m_textBox.get())
     {
-        const auto delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
-        if (delta > 0)
-        {
-            StepValueUp();
-        }
-        else if (delta < 0)
+        if (IsHyperScrollEnabled() && textBox.FocusState() != winrt::FocusState::Unfocused)
         {
-            StepValueDown();
+            const auto delta = args.GetCurrentPoint(*this).Properties().MouseWheelDelta();
+            if (delta > 0)
+            {
+                StepValueUp();
+            }
+            else if (delta < 0)
+            {
+                StepValueDown();
+            }
         }
     }
 }
 
 void NumberBox::StepValue(bool isPositive)
 {
-    auto newVal = Value();
-
-    if (isPositive)
-    {
-        newVal += StepFrequency();
-    }
-    else
-    {
-        newVal -= StepFrequency();
-    }
+    // Before adjusting the value, validate the contents of the textbox so we don't override it.
+    ValidateInput();
 
-    if (WrapEnabled())
+    auto newVal = Value();
+    if (!std::isnan(newVal))
     {
-        const auto max = Maximum();
-        const auto min = Minimum();
-
-        if (newVal > max)
+        if (isPositive)
         {
-            newVal = min;
+            newVal += StepFrequency();
         }
-        else if (newVal < min)
+        else
         {
-            newVal = max;
+            newVal -= StepFrequency();
         }
-    }
 
-    Value(newVal);
+        if (IsWrapEnabled())
+        {
+            const auto max = Maximum();
+            const auto min = Minimum();
+
+            if (newVal > max)
+            {
+                newVal = min;
+            }
+            else if (newVal < min)
+            {
+                newVal = max;
+            }
+        }
+
+        Value(newVal);
+    }
 }
 
 // Updates TextBox.Text with the formatted Value
 void NumberBox::UpdateTextToValue()
 {
     if (auto&& textBox = m_textBox.get())
     {
-        // Rounding the value here will prevent displaying digits caused by floating point imprecision.
-        const auto roundedValue = m_displayRounder.RoundDouble(Value());
+        const auto value = Value();
+        if (std::isnan(value))
+        {
+            // Display NaN as empty string
+            textBox.Text(L"");
```

Comment:
Is this redundant? I'm guessing it is needed because sometimes you'll be in this function while m_textUpdating is true so the DP setter wont update the textBox.  But I want to double check.

Reply:
I refactored it some, I was missing some things. It's probably better now.

Reply:
It still looks like you are setting the Text DP and the textBox.Text proprety.  In the TextPropertyChanged handler you set the textBox.Text proprety so I'm wondering if you need to be doing it here or not.

---
In reply to: [349339390](https://github.com/microsoft/microsoft-ui-xaml/pull/1656#discussion_r349339390) [](ancestors = 349339390)

Reply:
But since I set m_textUpdating, OnTextPropertyChanged won't actually do anything.